packages/model/src/CaretManagement/Caret/Caret.ts

```diff
@@ -0,0 +1,88 @@
+import type { Index } from '../../utils/index.js';
+import type { CaretSerialized, CaretEvent } from './types.js';
+import { CaretUpdatedEvent } from './types.js';
+import { EventBus } from '../../utils/index.js';
+
+/**
+ * Interface to extend EventTarget methods
+ */
+export interface Caret {
+  /**
+   * Adds CaretEvent listener
+   *
+   * @param event - type of event
+   * @param listener - listener
+   */
+  addEventListener<K extends CaretUpdatedEvent>(event: CaretEvent, listener: (event: K) => void): void;
+
+  /**
+   * Dispatches CaretUpdatedEvent
+   *
+   * @param event - event to dispatch
+   */
+  dispatchEvent(event: CaretUpdatedEvent): boolean;
+}
+
+/**
+ * Caret is responsible for storing caret index
+ */
+export class Caret extends EventBus {
+  /**
+   * Caret index
+   */
+  #index: Index | null = null;
```

Comment:
When caret index can be null? I think that in this case, we should delete the Caret instance because the index is null and this caret doesn't exist.

Reply:
I'm not sure for that at the moment. If user unfocuses the editor should we remove the caret or should we just set it to null. Also it could be null on creation